rosbag2_compression/src/rosbag2_compression/sequential_compression_writer.cpp

```diff
@@ -73,11 +73,18 @@ void SequentialCompressionWriter::compression_thread_fn()
             "Cannot compress message; Writer is not open!"};
   }
 
-  while (compression_is_running_) {
+  while (true) {
     std::shared_ptr<rosbag2_storage::SerializedBagMessage> message;
     std::string file;
     {
       std::unique_lock<std::mutex> lock(compressor_queue_mutex_);
```

Comment:
Shouldn't the main thread also grab this lock.  You are using it to gate the critical section check on the compression_is_running_ flag.

This current change looks like it makes the window smaller for the deadlock but does not remove it completely.

Reply:
Yes - you must be right. I've added the lock to `stop_compressor_threads` as well